autosklearn/metrics/__init__.py

```diff
@@ -491,45 +490,41 @@ def calculate_loss(
     task_type: int
         To understand if the problem task is classification
         or regression
-    metric: Scorer
-        Object that host a function to calculate how good the
+    metrics: Sequence[Scorer]
+        A list of objects that hosts a function to calculate how good the
         prediction is according to the solution.
     scoring_functions: List[Scorer]
         A list of metrics to calculate multiple losses
 
     Returns
     -------
-    float or Dict[str, float]
+    Dict[str, float]
         A loss function for each of the provided scorer objects
     """
-    score = calculate_score(
+    score = calculate_scores(
         solution=solution,
         prediction=prediction,
         task_type=task_type,
-        metric=metric,
+        metrics=metrics,
         scoring_functions=scoring_functions,
     )
-
-    if scoring_functions:
-        score = cast(Dict, score)
-        # we expect a dict() object for which we should calculate the loss
-        loss_dict = dict()
-        for metric_ in scoring_functions + [metric]:
-            # TODO: When metrics are annotated with type_of_target support
-            # we can remove this check
-            if metric_.name not in score:
-                continue
-            # maybe metric argument is not in scoring_functions
-            # so append it to the list. Rather than check if such
-            # is the case, redefining loss_dict[metric] is less expensive
-            loss_dict[metric_.name] = metric_._optimum - score[metric_.name]
-        return loss_dict
-    else:
-        rval = metric._optimum - cast(float, score)
-        return rval
-
-
-def calculate_metric(
+    scoring_functions = scoring_functions if scoring_functions else []
+
+    # we expect a dict() object for which we should calculate the loss
+    loss_dict = dict()
+    for metric_ in scoring_functions + list(metrics):
+        # TODO: When metrics are annotated with type_of_target support
+        # we can remove this check
+        if metric_.name not in score:
+            continue
```

Comment:
I guess this check is from before but I'm just curious as to when this happens, I guess it's when `calculate_scores` fails to actually use the metric or something goes wrong

Reply:
TBH I don't remember. I updated the text a bit to make it less confusing.